### PR TITLE
Module 4 - Default categories

### PR DESCRIPTION
**Pull request by sinpai**
* Added user creator service
* Added user default categories creator service
* Changed omniauth to user them
* Added rake task to polish old users

### CODE REVIEW

File: app/services/users/creator.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @user = User.create(
+      email: email,
+      password: Devise.friendly_token[0, 20],
+      confirmed_at: Date.current
+    )
+  end
+
+  def add_default_categories
+    Users::DefaultCategoriesCreator.new(@user).call
+    @user
```
Comment (AleksSenkou) [inline app/services/users/creator.rb]: @sinpai let's return `@user` from `#call` instead of `#add_default_categories`. `add_default_categories` doesn't say nothing about user, you can't expect that it returns user

File: app/services/users/default_categories_creator.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    default_categories_list.map { |name| create_category(name) }
+  end
+
+  private
+
+  def create_category(name)
+    user.categories.create(name: name, amount: 0, user_id: user.id)
+  end
+
+  delegate :categories, to: :user
```
Comment (AleksSenkou) [inline app/services/users/default_categories_creator.rb]: @sinpai why did you add `delegate` here and not at the end of the file? Can you try that?

File: app/services/users/default_categories_creator.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class Users::DefaultCategoriesCreator < Struct.new(:user)
+  def call
+    default_categories_list.map { |name| create_category(name) }
+  end
+
+  private
+
+  def create_category(name)
```
Comment (AleksSenkou) [inline app/services/users/default_categories_creator.rb]: @sinpai private methods with arguments say that a service is overloaded and you have to pass arguments to private methods

File: app/services/users/default_categories_creator.rb (hunk 2 above)
Comment (AleksSenkou) [reply app/services/users/default_categories_creator.rb]: @sinpai after this delegation you can use `categories` without `user`

Comment (AleksSenkou) [reply app/services/users/default_categories_creator.rb]: https://apidock.com/rails/Module/delegate

File: app/services/users/default_categories_creator.rb (hunk 3 above)
Comment (AleksSenkou) [reply app/services/users/default_categories_creator.rb]: @sinpai what do you think about:
```ruby
def call
  default_categories_list.map do |name|
    categories.create(name: name, amount: 0)
  end
end
```

File: app/services/users/default_categories_creator.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    user.categories.create(name: name, amount: 0, user_id: user.id)
+  end
+
+  delegate :categories, to: :user
+
+  def default_categories_list
+    %w[Transportation Food House Entertainment Health Other]
+  end
+end
```
Comment (AleksSenkou) [inline app/services/users/default_categories_creator.rb]: @sinpai tests for the service? Don't forget about it

**Review by AleksSenkou — COMMENTED** [review]
Suggestions

File: app/services/users/default_categories_creator.rb (hunk 4 above)
Comment (sinpai) [reply app/services/users/default_categories_creator.rb]: Will add them, thanks

**Review by AleksSenkou — APPROVED** [review]
LGTM 🥇
